tests/performance/like_and_match_pattern_caching.xml

```diff
@@ -0,0 +1,49 @@
+<!-- Simple benchmark to verify that compiled re2 patterns for LIKE/MATCH queries are internally cached -->
+
+<test>
+    <substitutions>
+        <substitution>
+           <name>numbers</name>
+           <values>
+               <value>numbers_mt(10000000)</value>
+           </values>
+        </substitution>
+    </substitutions>
+
+    <!-- LIKE/MATCH with non-const needle which cannot be cached (high distinctness) -->
+
+    <query>
+        select toString(number) as haystack, toString(number) || '%' as needle, like(haystack, needle)
```

Comment:
There should be an optimization for these types of patterns - make prefix comparison instead of running the regexp.

Reply:
True. We currently optimize "%pattern%" to a substring search. "pattern%" and "%pattern" could of course be made prefix/suffix searches. I could imagine that re2 already has similar optimizations but of course for const needles (the standard case), we can call the substring search on multiple values at once and save repeated calls into re2.

So *yes*, that is something worth looking into but I rather not like to squeeze this into this PR.